.github/workflows/build.yml

```
@@ -10,122 +10,78 @@ on:
       - grpc/gradle.properties
 
 jobs:
-  metadata:
-    runs-on: ubuntu-20.04
-    outputs:
-      grpc_version: ${{ steps.grpc_ver.outputs.release_version }}
-      grpc_changed: ${{ steps.filter.outputs.grpc }}
-      release_version: ${{ steps.ver.outputs.release_version }}
-    steps:
-      - uses: actions/checkout@v3
-      - uses: dorny/paths-filter@v2
-        id: filter
-        with:
-          filters: |
-            grpc:
-              - 'grpc/gradle.properties'
-      - name: Get gRPC release_version
-        id: grpc_ver
-        uses: christian-draeger/read-properties@1.1.1
-        with:
-          path: grpc/gradle.properties
-          properties: release_version
-      - name: Get release_version
-        id: ver
-        uses: christian-draeger/read-properties@1.1.1
-        with:
-          path: gradle.properties
-          properties: release_version
-  build-grpc-java:
-    runs-on: ubuntu-20.04
+  versions:
+    name: Extract app version
+    uses: th2-net/.github/.github/workflows/compound-prebuild-java-workflow.yml@main
+    changes:
+      name: Collect changed files
+      runs-on: ubuntu-20.04
+      outputs:
+        app: ${{ steps.filter.outputs.app }}
+        grpc: ${{ steps.filter.outputs.grpc }}
+      steps:
+        - uses: dorny/paths-filter@v2
+          id: filter
+          with:
+            filters: |
+              app:
+                - 'gradle.properties'
+              grpc:
+                - 'grpc/gradle.properties'
```

I don't think this  work when you use `uses` directive.
Probably you wanted this:
```suggestion
  changes:
      name: Collect changed files
      runs-on: ubuntu-20.04
      outputs:
        app: ${{ steps.filter.outputs.app }}
        grpc: ${{ steps.filter.outputs.grpc }}
      steps:
        - uses: dorny/paths-filter@v2
          id: filter
          with:
            filters: |
              app:
                - 'gradle.properties'
              grpc:
                - 'grpc/gradle.properties'
```